src/name.rs

```diff
@@ -77,20 +77,33 @@ impl<'a> From<DNSNameRef<'a>> for DNSName {
 #[derive(Clone, Copy)]
 pub struct DNSNameRef<'a>(untrusted::Input<'a>);
 
+/// An error indicating that a `DNSNameRef` could not built from the provided input.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct InvalidDNSNameError;
+
+impl std::fmt::Display for InvalidDNSNameError {
```

Comment:
We should use `core::fmt` instead of `std::fmt` here. I'll change this in a follow-up.

Reply:
> We should use `core::fmt` instead of `std::fmt` here. I'll change this in a follow-up.

Oh, sorry, I see that the build is actually broken in the current state so I can't merge this as-is.

Instead, change `std::fmt` to `core::fmt` everywhere in your PR and it should work.